noxfile.py

```diff
@@ -202,6 +213,15 @@ def coverage(session: nox.Session):
 @nox.session(python=False)
 def setup(session: nox.Session):
     """Set up the external environment."""
+    if session.interactive and not is_venv():
+        confirm = input(
+            "It looks like you are about to install the dependencies into your *global* python environment."
+            " This may overwrite other versions of the dependencies that you already have installed, including disnake itself."
+            " Consider using a virtual environment (virtualenv/venv) instead. Continue anyway? [y/N]"
+        )
+        if confirm.lower() != "y":
+            session.error("Canceled")
+
```

Comment:
do we want to use `cancelled` instead?

Reply:
sure
https://github.com/DisnakeDev/disnake/pull/628/commits/b514db93a66b948a8789f5d774d14a5f1761970a